Content.Server/Access/Systems/IdCardSystem.cs

```diff
@@ -74,5 +80,53 @@ private void UpdateEntityName(EntityUid uid, IdCardComponent? id = null)
                                 ("fullName", id.FullName),
                                 ("jobSuffix", jobSuffix));
         }
+
+        /// <summary>
+        ///     Attempt to find an ID card on an entity. This will look in the entity itself, in the entity's hands, and
+        ///     in the entity's inventory.
+        /// </summary>
+        public bool TryFindIdCard(EntityUid uid, [NotNullWhen(true)] out IdCardComponent? idCard)
+        {
```

Comment:
So this is basically the same functions as `AccessReaderSystem.FindAccessTags()`, but instead of looking for an `AccessComponent`, it's trying to find an `IdCardComponent`.

I'm not sure whether it's worth generalising this, as I don't think anything other than the `IdCardComponent` or `AccessComponent` would need to check those specific entities for a component:
- The user
- The held entity
- The id-card inventory slot
- Inside of a PDA's Id card slot

Are there even any entities with an `AccessComponent` that don't have an `IdCardComponent`? because if not, then only one of these is ever actually needed.

Reply:
would be nice if they could be merged in some way? but no idea what that would actually look like